examples/policies/kubernetes/clusterwide/health.yaml

```diff
@@ -0,0 +1,14 @@
+apiVersion: "cilium.io/v2"
+kind: CiliumClusterwideNetworkPolicy
+metadata:
+  name: "cilium-health-checks"
+spec:
+  endpointSelector:
+    matchLabels:
+      'reserved:health': ''
+  ingress:
+    - fromEntities:
+      - health
+  egress:
+    - toEntities:
+      - health
```

Comment:
I did some tests with a GKE cluster and I don't think this is correct. If I setup the cluster in `enable-policy=always` and load this policy, I get the following packet drops for the health traffic:
```
$ kubectl -n kube-system exec cilium-725nz -- cilium monitor -t policy-verdict | grep "action deny"
Policy verdict log: flow 0x0 local EP ID 12, remote ID remote-node, proto 1, ingress, action deny, match none, 10.164.0.44 -> 10.76.1.11 EchoRequest
Policy verdict log: flow 0x0 local EP ID 12, remote ID remote-node, proto 6, ingress, action deny, match none, 10.164.0.44:52576 -> 10.76.1.11:4240 tcp SYN
Policy verdict log: flow 0x0 local EP ID 12, remote ID remote-node, proto 6, ingress, action deny, match none, 10.164.0.44:52576 -> 10.76.1.11:4240 tcp SYN
Policy verdict log: flow 0x0 local EP ID 12, remote ID remote-node, proto 6, ingress, action deny, match none, 10.164.0.44:52576 -> 10.76.1.11:4240 tcp SYN
Policy verdict log: flow 0x0 local EP ID 12, remote ID remote-node, proto 6, ingress, action deny, match none, 10.164.0.44:52576 -> 10.76.1.11:4240 tcp SYN
Policy verdict log: flow 0x0 local EP ID 12, remote ID remote-node, proto 6, ingress, action deny, match none, 10.164.0.44:52576 -> 10.76.1.11:4240 tcp SYN
[...]
```
<br>

I think I know why you missed this :-)  According to [your previous comment](https://github.com/cilium/cilium/pull/15348#discussion_r594584445), you are using the development VMs (:+1:). Unfortunately, the development VM doesn't use exactly the same Cilium options as the Helm installation (more common way to install for real clusters). In particular, the development VM uses `enable-remote-node-identity=false` which leads to the automatic installation of the following rule in each endpoint:
```
POLICY   DIRECTION   LABELS (source:key[=value])   PORT/PROTO   PROXY PORT   BYTES   PACKETS       
Allow    Ingress     reserved:remote-node          ANY          NONE         5384    62 
```
Once we set `enable-remote-node-identity=true` (default in the GKE deployment for instance), that rule is removed and health check packets are dropped. So I think we should write the policy to work with the most common scenario (`enable-remote-node-identity=true`). Here is the policy I had to use to allow communication to the health endpoints:
```yaml
apiVersion: "cilium.io/v2"
kind: CiliumClusterwideNetworkPolicy
metadata:
  name: "health"
spec:
  endpointSelector:
    matchLabels:
      'reserved:health': ''
  ingress:
  - fromEntities:
    - remote-node
```

I've filed https://github.com/cilium/cilium/issues/15364, a new `good-first-issue`, to change the development VMs and prevent this from happening again in the future.

Reply:
Yes. I used development VM to conduct experiment. Thank you for pointing out!

Reply:
So to speak, should we have `remote-node` and `health` entity for healthcheck ingress/egress? like below
```
spec:
  endpointSelector:
    matchLabels:
      'reserved:health': ''
  ingress:
  - fromEntities:
    - health
    - remote-node
  egress:
  - toEntities:
    - health
    - remote-node
```

Reply:
I think only `remote-node` is needed. I didn't see any health-to-health traffic in my cluster.